### ISSUE TITLE
Allow having secondary Tag Manager container if site uses paired AMP

### ISSUE THREAD

**Issue by kmwilkerson** [google/site-kit-wp#413]
**Note:** The original report below includes multiple issues, but let's use this issue only to track "Issue 1" below.

Related resources:
* https://support.google.com/tagmanager/answer/9205783
* https://developers.google.com/tag-manager/api/v1/reference/accounts/containers/create (see usage context, we need to check whether this supports `amp` too)

Here's a screenshot (sorry, German) showing that there is a specific container type for AMP:
<img width="591" alt="Screenshot 2019-08-22 at 15 56 19" src="https://user-images.githubusercontent.com/3531426/63524144-c5731600-c4fb-11e9-9733-3d485d7df1b0.png">

## Bug Description

Issue 1: When the connection to GTM is made, it automatically inserts the GTM script across standard pages and AMP pages. However, the same GTM container is not meant to be used in both places, as GTM requires a unique container for AMP pages. When the Site Kit GTM connection is made, there needs to be a container selection for standard pages, and a second for AMP pages. Or an option to not insert on AMP.

Issue 1a: The Site Kit GTM script is missing the second half of the required snippet on AMP pages (that goes in the < body >).

Issue 1b: Additionally, there should probably be a note to let the user know to _remove_ the analytics settings they may have input _within_ the AMP plugin settings. As having AMP plugin analytics enabled + Site Kit GTM OR GA auto insert enabled will result in inflated data. This same risk applies to enabling the auto-insert of Site Kit GA snippet with GTM. GA snippet + GTM snippet should not coexist on a page together (when GA is an active variable in GTM), as that too could result in doubled data. 

## Steps to reproduce

**Info for issue 1:**
https://support.google.com/tagmanager/answer/9205783
The reason the same container shouldn't be used on both standard pages and AMP pages is due to AMP limitations when it comes to variables, tag types, and how those tags are fired. If the standard GTM container is used on AMP, there could be errors that prevent any data collection at all. 

**Steps to reproduce issue 1b:**

1. Enable [AMP plugin](https://wordpress.org/plugins/amp/).
2. Input GA information within AMP plugin analytics settings.
2. Connect GA (allow snippet placement) and GTM in Site Kit.
3. View the source on AMP pages -- see all 3 being called -- GA, GTM, + GA a second time from the AMP settings.
4. In real-time in GA, see 2 pageviews counted for this _single_ "real" pageview. (You won't see 3 due to the Site Kit GTM script missing its corresponding body tag.)

## Screenshots

*1a* Two parts are required for GTM on AMP pages. Site Kit is currently only inserting the < head > snippet.

<img width="707" alt="Screen Shot 2019-08-16 at 4 00 03 PM" src="https://user-images.githubusercontent.com/20666833/63197986-fdcbad80-c03e-11e9-8cb8-07f706067d1e.png">

*1b* Source code of AMP page with Site Kit GA + Site Kit GTM + AMP plugin analytics

<img width="1366" alt="Site Kit bug" src="https://user-images.githubusercontent.com/20666833/63197197-c9ef8880-c03c-11e9-97d6-ab3b27933b08.png">

## Additional Context

- AMP Plugin Version 1.2

**This issue is blocked by #432 and #433.**

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

During GTM setup, check if the site is using paired AMP. If yes:
* display text: "Looks like your site is using paired AMP. Please select your Tag Manager account and relevant containers below, the snippets will be inserted automatically to your site."
* add a third drop down for the AMP container. 
<img width="841" alt="Screenshot 2020-01-27 at 23 45 15" src="https://user-images.githubusercontent.com/13304791/73220834-a38f5680-415f-11ea-8edd-3c4ad10a82d8.png">
* all drop-downs should have an option either to set up a new account or set up a new container. 
* in the GTM section of Settings, add information about the AMP container ID as well.  

## Implementation Brief

**In Tag Manager setup, if `ampMode` is `secondary`:**
* Render an additional drop down for AMP Container which only lists containers where `usageContext` is `amp`
    - The added dropdown should also have a "Setup a new container" option
* Internally, the component will need to store `ampContainers` which it can query from the `containers` datapoint with `usageContext` of `amp`
    - Ideally a single datapoint could return containers for both usage contexts (consistent with Google API)
    - Let's update the `containers` datapoint to support multiple `usageContext`s
* Update the settings view for Tag Manager to show the AMP container ID

## Changelog entry

* Properly support paired AMP in Tag Manager module by allowing to select two different containers, one for `web` context and the other for `amp` context.

**Comment by felixarntz** [comment]
Thanks for the detailed report @kmwilkerson!

Since you reported multiple issues, we split these into a few:

1. We will use this issue to track implementation of dealing with Tag Manager for AMP differently than for non-AMP. In other words, Site Kit should ensure that, for AMP content, the Tag Manager container is an "AMP" container, and for non-AMP, the container is a "Web" container.
1a. As far as I can see, that is not an actual issue. We _do_ insert the second part too, just not at the beginning of the `body` tag, which however is not an actual problem. You can actually see this piece inserted in your screenshot in 1b - can you confirm?
1b. It is indeed a problem, we do not currently detect existing Tag Manager snippets in the page. I opened #433 to track that. Also related, our snippet detection does not work for sites that use AMP as a secondary version (like in your case), so for fixing that I opened #432.

I hope that addresses the problems!

**Comment by felixarntz** [comment]
IB ✅ 

@aaemnnosttv Just to clarify, in settings the containers that are displayed should also depend on the AMP mode. E.g. an AMP container should only be shown if primary or secondary AMP, a non-AMP container should only be shown if secondary AMP or no AMP.

**Comment by tofumatt** [comment]
QA ✅

Using paired mode I got the second container select and ID to appear in setup (not pictured, but it worked as expected), settings view, and settings edit:

<img width="611" alt="Screenshot 2020-02-25 21 13 10" src="https://user-images.githubusercontent.com/90871/75288221-c91d8780-5813-11ea-8c38-e839235bf38e.png">
<img width="1155" alt="Screenshot 2020-02-25 21 13 53" src="https://user-images.githubusercontent.com/90871/75288217-c6bb2d80-5813-11ea-8a73-5d526da59553.png">
